### PR TITLE
remove raiicontext and replace close() calls

### PR DESCRIPTION
**Pull request by anonrig**
We are diverging from the original implementation in terms of function signature, but this simplifies our implementation. I am planning on opening a pull-request to Node.js as well to keep the implementations in sync with each other.

### CODE REVIEW

File: src/workerd/api/node/zlib-util.c++ [hunk 1]
```diff
@@ -397,7 +397,7 @@ kj::Maybe<CompressionError> ZlibContext::setParams(int _level, int _strategy) {
   return kj::none;
 }
 
-void ZlibContext::close() {
+ZlibContext::~ZlibContext() {
```
Comment (jasnell) [inline src/workerd/api/node/zlib-util.c++:400]: I see these are marked as `KJ_DISALLOW_COPY` but what if these are `kj::mv`'d? Will the destructor on the moved-from original be safe? Or should these be `KJ_DISALLOW_COPY_AND_MOVE`?

File: src/workerd/api/node/zlib-util.c++ [hunk 2]
```diff
@@ -655,7 +655,7 @@ BrotliEncoderContext::BrotliEncoderContext(ZlibMode _mode): BrotliContext(_mode)
   state = kj::disposeWith<BrotliEncoderDestroyInstance>(instance);
 }
 
-void BrotliEncoderContext::close() {
+BrotliEncoderContext::~BrotliEncoderContext() {
   auto instance = BrotliEncoderCreateInstance(alloc_brotli, free_brotli, alloc_opaque_brotli);
```
Comment (jasnell) [inline src/workerd/api/node/zlib-util.c++:659]: I meant to ask about this before but forgot... why is the destructor creating a new instance here and assigning to state?

File: src/workerd/api/node/zlib-util.h [hunk 3]
```diff
@@ -90,15 +90,14 @@ struct CompressionError {
   int err;
 };
 
-// TODO(soon): See if RAII support can be added directly to this class, and we can mark it final
-class ZlibContext {
+class ZlibContext final {
 public:
   explicit ZlibContext(ZlibMode _mode): mode(_mode) {}
   ZlibContext() = default;
+  ~ZlibContext();
```
Comment (jasnell) [inline src/workerd/api/node/zlib-util.h:97]: By convention we use `noexcept(false)` for destructors.

**Review by npaun — APPROVED** [review]

File: src/workerd/api/node/zlib-util.c++ (hunk 1 above)
Comment (anonrig) [reply src/workerd/api/node/zlib-util.c++]: You're right. I've added them all. Thank you!

File: src/workerd/api/node/zlib-util.h (hunk 3 above)
Comment (anonrig) [reply src/workerd/api/node/zlib-util.h]: Yes, actually I realized that we don't even need them...

File: src/workerd/api/node/zlib-util.c++ (hunk 2 above)
Comment (anonrig) [reply src/workerd/api/node/zlib-util.c++]: I've removed them all. I had an eye opening moment thanks to you. I've removed the destructors.

**Review by jasnell — APPROVED** [review]

**Pull request merged**